app/services/remote_tagging_service.rb

```diff
@@ -75,7 +79,7 @@ def check_for_exceptions(response, message_prefix)
     if response.status == 403
       raise Exceptions::NotAuthorizedError, response.reason_phrase
     else
-      raise "#{message_prefix} #{response.reason_phrase}" unless VALID_HTTP_CODES.include?(response.status)
+      raise TaggingError.new("#{message_prefix} #{response.reason_phrase}") unless VALID_HTTP_CODES.include?(response.status)
```

Comment:
Can we detect connection and timeout errors like how we make Kie api calls? If yes, we can mark them service_unavailable instead of bad request.

Reply:
I went through the underlying `faraday` gem used by remote tagging, it's hard to differentiate the `connection` or `timeout` errors based on the response status. So the answer is no

Reply:
Changed to catch Faraday exceptions directly